### ISSUE TITLE
Module Requirements (2008R2 core)

### ISSUE THREAD

**Issue by gmitch64** [Crypt32/PSPKI#29]
I'm looking at the module requirements

> This module can run on any of the specified operating system:
> 
> Windows Server 2008*/2008 R2/2012/2012 R2
> Windows Vista**/7**/8**/8.1**/10**
> * — Server Core installation is not supported.
> 
> ** — with installed RSAT (Remote System Administration Tools)

So, according to this,  the module will NOT run on 2008 Core, but SHOULD run on 2008R2 Core. I'm saying should, as code that I have working on 2008R2 full GUI is throwing a CAPI unavailable error on R2 core.

I have NetFX2-Server Core, plus NDP451 installed on the server (as well as WMF 5.1).

Is there something that I'm missing that I need to install, or does the module just not run on 2008R2 core?

Thanks.

Graham

**Comment by Crypt32** [comment]
For module installation, you need to install ADCS management tools (RSAT).

**Comment by gmitch64** [comment]
But server core doesn't support RSAT installation (at least not on 2008R2). You get an 'update is not applicable to your computer error'

**Comment by Crypt32** [comment]
Despite the error (about CAPI), client-side functionality (commands in Client folder) are still working, since they don't depend on server CAPI.

I updated module installation requirements to reflect actual state: you can install the module on server core, but with limited functionality. In order to get full functionality, full GUI w/RSAT is necessary. CAs on server core are fully supported.

**Comment by gmitch64** [comment]
The install works fine on 2008R2, I get the error when trying to generate a self signed certificate for using with WinRM.

$Cert = New-SelfSignedCertificateEx -Subject CN=$FQDN -StoreLocation LocalMachine -EKU "Server Authentication"

I've just seen your update to the Readme.MD file. So basically looks like I'm out of luck on any of the Core installs. Thanks for the update on that. Looks like I may have to see what I can do with makecert.

**Comment by Crypt32** [comment]
> I've just seen your update to the Readme.MD file. So basically looks like I'm out of luck on any of the Core installs.

not really. `New-SelfSignedCertificateEx` perfectly works on server core (except Windows Server 2008) despite CAPI errors. CAPI error means that remote ADCS server management is blocked, but client-specific stuff, like certificate generation is supported.

**Comment by Crypt32** [comment]
In order to make it working on Server Core, delete `Server` folder from module installation package and the error will go away.

**Comment by gmitch64** [comment]
Absolutely perfect. Just tried, and it works like a charm. 

Is this something you can change on the install/load, so that the module loads and the client functions work on core? Do you want me open a new issue, or do you want to attach it to this one?

Graham

**Comment by gmitch64** [comment]
Thanks for your help. I've opened issue #30. Hopefully it's not too wordy.
I'll close this issue.
